### DISCUSSION TITLE
Unable to change git branch immediately

### DISCUSSION THREAD

**Discussion by Louis-Ni** [JanDeDobbeleer/oh-my-posh#3080] [Troubleshoot]
it's weird when I change the git branch from other Git management tools like Fork but it won't change in the current terminal input box. however, i should leave the current dir and re-enter then it shows correctly
see the picture below
<img width="765" alt="image" src="https://user-images.githubusercontent.com/56672404/201689681-4de36c1b-b342-4b35-b760-5779b5dc6ffe.png">

**Comment by JanDeDobbeleer** [comment]
@Louis-Ni the prompt is printed only on enter after the command exits. There's no functionality that watches the directory and reloads the prompt on changes.
